aima-core/src/main/java/aima/core/probability/BeliefState.java

```diff
@@ -0,0 +1,6 @@
+package aima.core.probability;
+
+public interface BeliefState<Action,Percept> {
+    void update(Action action,Percept percept);
+
+}

```

Comment:
You don't require this interface for the AIMA3e branch. You can build DTAgent via existing interfaces.

Reply:
I tried to find but can't find any. Also, this gives exclusivity to this algorithm as using BeliefState it gets much closer to the pseudocode.